packages/flutter_tools/test/commands.shard/hermetic/http_host_validator_test.dart

```diff
@@ -0,0 +1,266 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/doctor_validator.dart';
+import 'package:flutter_tools/src/http_host_validator.dart';
+
+import '../../src/common.dart';
+import '../../src/fake_http_client.dart';
+import '../../src/fakes.dart';
+
+// The environment variables used to override some URLs
+const String kTestEnvPubHost = 'https://pub.flutter-io.cn';
+const String kTestEnvGCloudHost = 'https://storage.flutter-io.cn';
+const Map<String, String> kTestEnvironment = <String, String>{
+  'PUB_HOSTED_URL': kTestEnvPubHost,
+  'FLUTTER_STORAGE_BASE_URL': kTestEnvGCloudHost,
+  'FLUTTER_DOCTOR_HOST_TIMEOUT': '1',
+};
+
+void main() {
+  group('http host validator', () {
+    const List<String> osTested = <String>['windows', 'macos', 'linux'];
+
+    group('no env variables', () {
+      test('all http hosts are available', () async {
```

Comment:
Sorry for not noticing this earlier, but it would be good to try to make these tests more hermetic by using `testWithoutContext`, which will ensure that the code under test doesn't access potentially non-fake/non-mock members of the global context.